pkg/pulumiyaml/packages/file.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2024, Pulumi Corporation.  All rights reserved.
+
+// Package packages contains utilities for working with Pulumi package lock files.
+package packages
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/blang/semver"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+	"gopkg.in/yaml.v3"
+)
+
+type ParameterizationDecl struct {
+	// Name is the name of the parameterized package.
+	Name string `yaml:"name"`
+	// Version is the version of the parameterized package.
+	Version string `yaml:"version"`
+	// Value is the value of the parameter. Value is a base64 encoded byte array, use SetValue and GetValue to manipulate the actual value.
+	Value string `yaml:"value"`
+}
+
+func (p *ParameterizationDecl) GetValue() ([]byte, error) {
+	return base64.StdEncoding.DecodeString(p.Value)
+}
+
+func (p *ParameterizationDecl) SetValue(value []byte) {
+	p.Value = base64.StdEncoding.EncodeToString(value)
+}
+
+type PackageDecl struct {
+	// PackageDeclarationVersion is the version of the package declaration file.
+	PackageDeclarationVersion int `yaml:"packageDeclarationVersion"`
+	// Name is the name of the plugin.
+	Name string `yaml:"name"`
+	// Version is the version of the plugin.
+	Version string `yaml:"version,omitempty"`
+	// PluginDownloadURL is the URL to download the plugin from.
+	DownloadURL string `yaml:"downloadUrl,omitempty"`
```

Comment:
Nit/question: do we have any prior art when it comes to acronyms in YAML? E.g. do we favour `downloadUrl` or `downloadURL`? The only one I know is `backend: url:`, but that's the first word so it's lowercase :thinking:

Reply:
As with all things at Pulumi it seems inconsistent. In schema files we have "logoUrl" but "pluginDownloadURL". In deployments we have "roleArn" but in state we have "resourceID".

So flip a coin, but "Url" feels more YAML right to me, I don't know why.